Introduce a Polish translation of EventFahrplan

rotsq opened the pull request:
# Description
This pull request introduces a Polish translation of EventFahrplan.

# Before
No Polish translation.

# After
Polish translation.

Resolves #173

johnjohndoe commented:
@rotsq Awesome. Thank you for the effort. I will take a look tomorrow.

johnjohndoe commented on app/src/main/res/values-pl/strings.xml:
```diff
@@ -0,0 +1,245 @@
+        <item quantity="zero">Brak prelegentów</item>
+        <item quantity="one">Prelegent: <xliff:g example="Jane Doe" id="name">%s</xliff:g></item>
+        <item quantity="other">Prelegenci: <xliff:g example="Jane Doe, John Doe" id="names">%s</xliff:g></item>
+    </plurals>
```
Android Studio (Lint) points out the following for plurals in Polish:

> For locale "pl" (Polish) the following quantities should also be defined: few, many

![plurals](https://user-images.githubusercontent.com/144518/145777800-1a79cf13-b29b-4504-8d68-aa57e8b81850.png)

Can you please check and probably complete the plural occurrences in this file if the suggestion fit to the language. Please be critical - Lint might not know Polish as good as you do.

Please keep `zero` here because we want another text.

rotsq replied:
OK, I added 'few' and 'many'.

johnjohndoe replied:
Can you check the other plurals in the file, too please?

johnjohndoe replied:
It might not make sense to put few/many in the particular context. It is hard for me to tell with the Polish translation. The initial reason for plurals is the different words/word endings.

rotsq replied:
OK, few and many fitted also here: `schedule_changes_dialog_number_of_sessions`, so I did extend it by these two cases.

johnjohndoe commented on app/src/ccc36c3/res/values-pl/strings.xml:
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <string name="app_name">Grafik 36C3</string>
+    <string name="app_hardcoded_subtitle">27-30 grudnia 2019, Lipsk, Leipziger Messe</string>
```
Android Studio (Lint) suggests the following here:

> Replace "-" with an "en dash" character (–, `&#8211;`) ?

![dash](https://user-images.githubusercontent.com/144518/145778377-c8e2defd-8b81-479d-acad-c2c2cc34a0d0.png)

Please apply the suggestion (ALT + RETURN) here and in the `values-pl/strings.xml` file. Please notet that Lint only highlights such cases in files of the currently selected build flavor.

![image](https://user-images.githubusercontent.com/144518/145779327-faa338d2-e227-441f-aecd-1905a50911a9.png)

rotsq replied:
`-` replaced with `–` (and I agree that this is a better character). Bear in mind that this was taken from the English source, so probably a correction would be also needed right there, but this goes slightly outside of the Polish translation scope, so I am not touching the `values` folder.

rotsq replied:
Plus I also changed the location of the `Logo` text:
```
<string name="about_logo_content_description">
        36C3: Resource exhaustion Logo
</string>
```
with
```
<string name="about_logo_content_description">
        Logo 36C3: Resource exhaustion
</string>
```
as I overlooked that at the very beginning (in Polish the text `logo` goes before what it reflects).

@johnjohndoe, if you are happy with the changes you can resolve this conversation, as I believe you are the right person to do so, not me.

johnjohndoe reviewed (changes requested):
Dear @rotsq. I took a look at your work. It looks very good. :heart_eyes_cat: The app runs fine on my device with Polish chosen.

I just found a few minor spots to be corrected. Please take a look.

I am looking forward to do the final check.

johnjohndoe commented:
Here is an example screenshot of where the texts you translated are used:

![screenshot](https://user-images.githubusercontent.com/144518/145975372-27745c0d-d2d9-43c0-80ed-8b2744faaae6.png)

johnjohndoe commented on app/src/main/res/values-pl/strings.xml:
```diff
@@ -0,0 +1,249 @@
+
+    <!-- Schedule changes dialog -->
+    <string name="schedule_changes_dialog_title">Aktualizacja grafiku</string>
+    <string name="schedule_changes_dialog_updated_to_text">Zaktualizowano grafik do wersji\\u0020</string>
```
There is one backslash to much. I will change `\\u0020` to `\u0020`.

rotsq replied:
Is this the only place where its used? It is not in nominativ (and I need to change it), I thought it is.

johnjohndoe replied:
I don't understand to which word you are referring as nominativ here. Did you comment on the right place?

rotsq replied:
> Here is an example screenshot of where the texts you translated are used:
> 
> ![screenshot](https://user-images.githubusercontent.com/144518/145975372-27745c0d-d2d9-43c0-80ed-8b2744faaae6.png)

I was referring to the nominativ regarding the screenshot. Fixing itnow , since I thought that this was in nominativ, but it is not. And I'm looking at other changes introduced earlier today. In one hour I should be ready.

johnjohndoe commented on app/src/main/res/values-pl/strings.xml:
```diff
@@ -0,0 +1,249 @@
+    <string name="schedule_changes_dialog_title">Aktualizacja grafiku</string>
+    <string name="schedule_changes_dialog_updated_to_text">Zaktualizowano grafik do wersji\\u0020</string>
+    <string name="schedule_changes_dialog_changed_new_cancelled_text">. Są zmiany w <xliff:g example="3 wykładach" id="num_changes">%1$s</xliff:g>, <xliff:g example="2 są" id="num_new">%2$s</xliff:g> nowe, <xliff:g example="1 jest" id="num_canceled">%3$s</xliff:g> anulowany.</string>
+    <string name="schedule_changes_dialog_affected_text">Zmiany dotyczą <xliff:g example="2" id="num_marked">%1$d</xliff:g> Twoich ulubionych wykładów.</string>
```
As it is part of the sentence `Twoich` has to be lower case I believe. I will change this.

johnjohndoe replied:
:heavy_check_mark: Done.

johnjohndoe commented on app/src/main/res/values-pl/strings.xml:
```diff
@@ -0,0 +1,249 @@
+        <item quantity="one"><xliff:g example="1" id="count">%d</xliff:g> wykład</item>
+        <item quantity="other"><xliff:g example="9" id="count">%d</xliff:g> wykłady</item>
+        <item quantity="few">Kilka wykładów</item>
+        <item quantity="many">Wiele wykładów</item>
```
As it is part of the sentence `Kilka` and `Wiele` have to be lower case I believe. I will change this.

johnjohndoe replied:
:heavy_check_mark: Done.

johnjohndoe commented on app/src/rc3/res/values-pl/strings.xml:
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <string name="app_name">Grafik rC3</string>
+    <string name="app_hardcoded_subtitle">27&#8211;30 grudnia 2020, online</string>
```
I prefer to use `–` instead of the HTML entity code. I will change this here and in the `ccc36c3` file.

johnjohndoe reviewed (approved):
Thank you @rotsq for this very complete translation!! I checked the corrections again and just made a few corrections as pointed out in my comments.

I squashed your commits and rebased your branch onto the latest commit on `master`.

I will soon merge it if there is nothing left pointed out by you. :hourglass:

rotsq commented:
> I squashed your commits and rebased your branch onto the latest commit on `master`.
> 
> I will soon merge it if there is nothing left pointed out by you. hourglass

OK, so I looked for the source of plural definiton for different languages and in Polish we'll need `one`, `few` and `many` (not `other`): [source](https://unicode-org.github.io/cldr-staging/charts/latest/supplemental/language_plural_rules.html). I did change it according to these rules, but there is one more thing we need to cover if we want to be absolutely correct regarding the translation, here:

```<string name="schedule_changes_dialog_changed_new_cancelled_text">. Są zmiany w <xliff:g example="3 wykładach" id="num_changes">%1$s</xliff:g>, <xliff:g example="2 są" id="num_new">%2$s</xliff:g> nowe, <xliff:g example="1 jest" id="num_canceled">%3$s</xliff:g> anulowany.</string>```

Actually, `nowe` and `anulowany` would also need to be adequately modified based on the quantity:
- for `one` it would be `nowy` and `anulowany`
- for `few` it would be `nowe` and `anulowane`
- for `many` it would be `nowych` and `anulowanych`

What can we do here, @johnjohndoe? It will surely break your convention, but maybe you've got an idea.

johnjohndoe commented:
@rotsq I think we should leave it as almost perfect for now. We can take your finding to a follow-up issue or pull request. There I will see if I can change the text structure to enable the change. Okay?

rotsq commented:
Sure, I'm absolutely fine with that.

johnjohndoe commented:
> Actually, `nowe` and `anulowany` would also need to be adequately modified based on the quantity:

@rotsq I addressed your suggestion in #437. Can you please have a look if I implemented it correctly.

rotsq commented:
I commented on that in #437 referring to a particular line. It's almost great!

johnjohndoe commented:
@rotsq I cannot see you comment. Did you submit it?

rotsq commented:
@johnjohndoe, I thought that the review option in the code works differently and is public, and apparently, it does not work this way. Now I made an [in-line comment](https://github.com/EventFahrplan/EventFahrplan/pull/437/commits/f2f0348dfe59ab36892330b2ec488e162dd16385#r781501718), hopefully, visible for all.

johnjohndoe commented:
Happy new year @rotsq! As you probably saw I started managing translations at [Crowdin](https://crowdin.com/project/eventfahrplan). I like to invite you to join there if you like :)
